frontend/pkg/frontend/frontend.go

```diff
@@ -511,11 +504,21 @@ func (f *Frontend) ArmSubscriptionPut(writer http.ResponseWriter, request *http.
 		doc.Subscription = &subscription
 	}
 
+	// Log differences
+	f.logSubscriptionDifferences(doc.Subscription, &subscription)
```

Comment:
A small suggestion: maybe this line could be 
```go
logSubscriptionDifferences(f.logger, doc.Subscription, &subscription)
```
as we don't need anything other than the logger inside the function. Also, the frontend has a lot of methods defined right now.